cpp/src/io/parquet/reader_impl_chunking.hpp

```diff
@@ -30,58 +30,101 @@ struct file_intermediate_data {
   // all row groups to read
   std::vector<row_group_info> row_groups{};
 
-  // all chunks from the selected row groups. We may end up reading these chunks progressively
-  // instead of all at once
+  // all chunks from the selected row groups.
   std::vector<ColumnChunkDesc> chunks{};
 
   // an array of offsets into _file_itm_data::global_chunks. Each pair of offsets represents
   // the start/end of the chunks to be loaded for a given pass.
   std::vector<std::size_t> input_pass_row_group_offsets{};
+
   // row counts per input-pass
   std::vector<std::size_t> input_pass_row_count{};
 
+  size_t _current_input_pass{0};  // current input pass index
+  size_t _output_chunk_count{0};  // how many output chunks we have produced
+
   // skip_rows/num_rows values for the entire file. these need to be adjusted per-pass because we
   // may not be visiting every row group that contains these bounds
   size_t global_skip_rows;
   size_t global_num_rows;
+
+  size_t num_passes()
+  {
+    return input_pass_row_group_offsets.size() == 0 ? 0 : input_pass_row_group_offsets.size() - 1;
+  }
 };
 
 /**
- * @brief Struct to identify the range for each chunk of rows during a chunked reading pass.
+ * @brief Struct to identify a range of rows.
  */
-struct chunk_read_info {
+struct row_range {
+  size_t skip_rows;
+  size_t num_rows;
```

Comment:
"No rows".  The more I see `skip_rows`/`num_rows` the more I wish we used the more idiomatic `start_row`/`end_row` since we are always working on absolute indices any way (skip_rows is always an absolute value within the selected row groups, regardless of pass, subpass, output chunk, etc).  

It's probably worth a followup when someone has a bunch of free time :)

Reply:
I support this. Let's call `start_row` and `end_row`.